instrumentation/executors/bootstrap/src/main/java/io/opentelemetry/javaagent/bootstrap/executors/ContextPropagatingRunnable.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.bootstrap.executors;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+
+public final class ContextPropagatingRunnable implements Runnable {
+
+  public static boolean shouldDecorateRunnable(Runnable task) {
+    // We wrap only lambdas' anonymous classes and if given object has not already been wrapped.
+    // Anonymous classes have '/' in class name which is not allowed in 'normal' classes.
+    // note: it is always safe to decorate lambdas since downstream code cannot be expecting a specific runnable implementation anyways
+    return task.getClass().getName().contains("/") && !(task instanceof ContextPropagatingRunnable);
```

Comment:
This can handle the "lambda" situation, but how about the singleton task?

Reply:
This PR does not fix that. We're planning to redesign the `executors` instrumentation and fix that issue across the board; but that might take a while and will not be included in the next release (which is in ~ 5 days).